litgpt/finetune/adapter.py

```diff
@@ -75,6 +75,8 @@ def setup(
         logger_name: The name of the logger to send metrics to.
         seed: The random seed to use for reproducibility.
     """
+    if not checkpoint_dir.is_dir() and not checkpoint_dir.parts[0] == "checkpoints":
+        checkpoint_dir = "checkpoints" / checkpoint_dir
```

Comment:
Maybe just
```python
If not str(checkpoint_dir).startswith("checkpoints/"):
...
```
?

Btw, can it be a case when the path doesn't start with "checkpoints" but still such a folder exists?

Reply:
Keep forgetting about windows all the time 😔. So your approach with "parts" is better.

Reply:
I think the `if not checkpoint_dir.is_dir()` check should always be there, otherwise you can't pass a custom folder like `/tmp/my_custom_dir/llama3`.

And regarding  `str(checkpoint_dir).startswith("checkpoints/"):`, I think this won't work on windows since the slashes work different there, so I used `checkpoint_dir.parts[0] == "checkpoints"`, but please correct me if I'm wrong